controllers/eventController.js

```diff
@@ -57,6 +57,52 @@ const getEvent = (req, res, next) => {
 	Event.findOne({ _id: eventId }, (err, doc) => callback(err, doc, res, next));
 };
 
+const getWishlist = async (req, res, next) => {
+	try {
+		const userId = req.params.userId;
+		const user = await User.findOne({ _id: userId });
+		res.status(200);
+		res.send(user.wishlist);
+	} catch (error) {
+		next(error);
+	}
+};
+
+const addToWishlist = async (req, res, next) => {
+	try {
+		const userId = req.params.userId;
+		const user = await User.findOne({ _id: userId });
+		const eventId = req.params.eventId;
+		const event = await Event.findOne({ _id: eventId });
+		if (!user.wishlist.includes(event._id)) {
+			user.wishlist.push(event);
+			await user.save();
+			res.status(200);
+			res.send("added to wishlist");
+		}
+	} catch (error) {
+		next(error);
+	}
+};
+
+const removeFromWishlist = async (req, res, next) => {
+	try {
+		const userId = req.params.userId;
+		const user = await User.findOne({ _id: userId });
+		const eventId = req.params.eventId;
+		const event = await Event.findOne({ _id: eventId });
+		if (user.wishlist.includes(event._id)) {
+			return;
+		}
+		user.wishlist.splice(event._id, 1); // ??
```

Comment:
I do think this will work, but the Mongoose documentation had an example of something that looks quite nice at https://mongoosejs.com/docs/subdocs.html#removing-subdocs but I have never tried it :thinking: 

Perhaps something like this would do it too

```
user.wishlist.id(event._id).remove();
```